orchestration/hca_orchestration/solids/create_snapshot.py

```diff
@@ -72,6 +85,45 @@ def get_completed_snapshot_info(context: AbstractComputeExecutionContext, job_id
     yield Output(snapshot_info_dict['id'])
 
 
+@solid(
+    config_schema=Field(Permissive({"validate_snapshot_name": Field(bool, default_value=True, is_required=False)})),
+    required_resource_keys={'data_repo_client', 'snapshot_config', 'hca_manage_config', 'data_repo_service'},
+)
+def get_snapshot_from_project(context: AbstractComputeExecutionContext) -> Any:
+    """
+    Use the snapshot_name to get the associated snapshot_id from TDR,
+    for the snapshot created in the previous pipeline step - cut_snapshot.
+    If there is no matching snapshot or dataset - fail
+    If the snapshot_name does not end in the release_tag fail - so that we know we are using the correct snapshot
+    """
+    data_repo_service: DataRepoService = context.resources.data_repo_service
+    dataset_name = context.resources.snapshot_config.dataset_name
+    snapshot_name = context.resources.snapshot_config.snapshot_name
+    release_tag = context.resources.snapshot_config.qualifier
+    dataset = data_repo_service.find_dataset(dataset_name)
+
+    # we need the dataset to get the billing profile id, which is needed to query (enumerate) the snapshot
+    if not dataset:
+        raise Failure(f"Snapshot not found for dataset name [dataset_name={dataset_name}]")
+    if not snapshot_name:
+        raise Failure(f"Snapshot name not found for snapshot name [snapshot_name={snapshot_name}]")
+    else:
+        if not release_tag:
+            raise Failure(f"Release tag not found for release tag [release_tag={release_tag}]. This is required.")
+        if not snapshot_name.endswith(release_tag):
+            raise Failure(f"Snapshot name does not end in current release tag [snapshot_name={snapshot_name}], \
+            [release_tag={release_tag}].")
+        else:
+            response = context.resources.data_repo_client.enumerate_snapshots(filter=dataset_name)
+            print(f"create_snapshot get_snapshot_from_project response = {response}")
+            try:
+                snapshot_id = response.items[0].id
+                print("snapshot_id = ", snapshot_id)
+                return snapshot_id
+            except IndexError:
+                raise ValueError("The provided dataset name returned no results.")
```

Comment:
It'd be cleaner to check for len(response.items) == 1 and raise instead of catching index error rather than relying on an exception for the control flow.

Reply:
So this instead?
```
response = context.resources.data_repo_client.enumerate_snapshots(filter=dataset_name)
if not len(response.items) == 1
    raise Failure(f"There is more than one snapshot matching this dataset_name")
snapshot_id = response.items[0].id
return snapshot_id
```

Reply:
Yeah, I'd tweak the conditional to `if len(response.items) != 1`